### DISCUSSION TITLE
How to choose from multiple printer configurations on result screen / gallery

### DISCUSSION THREAD

**Discussion by PixelAlarm** [photobooth-app/photobooth-app#283] [General]
Hi,

Offtopic:
I have a printer that prints 20x15cm. Possible media formats are 1x full-size: 20x15cm or 2x half-size: 10x15cm.

Actual question:
My idea was to have two printer routines: 1x full-size or 2x half-size.
Now I set up two routines, but I cannot choose between configurations on result screen or from gallery.
There is only one button that will use the first configuration.

Is there a way to choose between the routines?

**Comment by mgineer85** [comment]
Hi,
not possible currently. It is just prepared to use it that way but the implementation is not finished.
In #276 there was also the idea to grneralize even further and change it to like a “print and share” functionality.

If you have ideas how the implementation would look like best feel free to add descriptions to the issues. Or tackle the issues and send a pull request :)

**Comment by PixelAlarm** [comment]
Hi,
I will think about the choice-options and return here.

For now, I'm using a small bash script (since I have to deal with different aspect ratios) but no user-choice:
- Collages are printed in full-size.
- Pictures are send twice and printed in half-size twice (printer cuts the paper itself).

For this purpose I configured the same printer twice with different settings.

**Reply by mgineer85** [reply]
You mind share the script? Could be helpful for others and me to understand in detail the requirements and tested solutions

**Reply by PixelAlarm** [reply]
Sure, here you go:

```
#!/bin/bash
#Needs imagemagick:
#sudo apt install imagemagick -y

PICTURE="$1"
height="$(identify -format '%h' $PICTURE)"

TMP_COLLAGE="/home/foto/photobooth-data/media/Collagendruck.jpg"
TMP_PHOTO="/home/foto/photobooth-data/media/Fotodruck.jpg"

#Check if Collage or Image (Collage is 4200x3150)
if [[ $height == 3150 ]]; then
    #Collage
    lp -d DI-RS1 -o landscape -o fit-to-page $PICTURE
else
    #Image
    convert $PICTURE $PICTURE -append $TMP_PHOTO
    lp -d DI-RS1-2 -o orientation-requested=3 -o fit-to-page $TMP_PHOTO
fi
```

**Reply by Tyfab56** [reply]
can you explain where you put this script ?

**Reply by GitNees** [reply]
> Sure, here you go:
> 
> ```
> #!/bin/bash
> #Needs imagemagick:
> #sudo apt install imagemagick -y
> 
> PICTURE="$1"
> height="$(identify -format '%h' $PICTURE)"
> 
> TMP_COLLAGE="/home/foto/photobooth-data/media/Collagendruck.jpg"
> TMP_PHOTO="/home/foto/photobooth-data/media/Fotodruck.jpg"
> 
> #Check if Collage or Image (Collage is 4200x3150)
> if [[ $height == 3150 ]]; then
>     #Collage
>     lp -d DI-RS1 -o landscape -o fit-to-page $PICTURE
> else
>     #Image
>     convert $PICTURE $PICTURE -append $TMP_PHOTO
>     lp -d DI-RS1-2 -o orientation-requested=3 -o fit-to-page $TMP_PHOTO
> fi
> ```

How would this work? I dont see anywhere images called collagendruck.jpg after I made some collages. its all random names. Also where do you run/put the script?

**Reply by PixelAlarm** [reply]
I‘m not at my setup right now.
Must be something like:

1. Make a file print.sh in /home/*youruser* with the code above
2. Install imagemagick as explained
3. Grant execution permission for print.sh
4. Use this as Print command:
„/home/*youruser*/print.sh {filename}“
